cli/cli_test.go

```diff
@@ -35,6 +35,16 @@ type cliTest struct {
 	*server.TestServer
 }
 
+// Stop the server, or wait until stopping is done
+// if it was already initiated by the caller.
+func (c *cliTest) cleanup(wasStopped *bool) {
+	if *wasStopped {
```

Comment:
Well what if the tests would also actually test something and do stuff like `t.FailNow()`?
An alternative I've considered was making Server.Stop() safe to call multiple times, but it didn't look trivial and I wasn't sure I liked it anyway.